src/routes/safe/store/actions/fetchTransactions.js

```diff
@@ -164,11 +168,25 @@ export const loadSafeTransactions = async (safeAddress: string) => {
   return Map().set(safeAddress, List(txsRecord))
 }
 
-export default (safeAddress: string) => async (dispatch: ReduxDispatch<GlobalState>) => {
+
+export default (safeAddress: string, userAddress: string, granted: boolean, onNotificationClosesCb: Function) => async (dispatch: ReduxDispatch<GlobalState>) => {
   try {
     const transactions: Map<string, List<Transaction>> = await loadSafeTransactions(safeAddress)
+    dispatch(addTransactions(transactions))
+    const awaitingTransactions = getAwaitingTransactions(transactions, userAddress, granted)
```

Comment:
I don't think that `fetchTransactions` action is a suitable place for this because we only need to show the notification once for each safe and complicating an action which runs every 5s with additional logic is not the best thing imo. I'd put this logic in a redux middleware that would fire a notification on `ADD_TRANSACTIONS` action if there is a transaction which needs user's confirmation. To keep track of safes with already displayed notifications, you could make a `Set` which would store a notification key.

Reply:
Good idea, now I did that, please check it